todo/src/main.rs

```diff
@@ -24,36 +38,57 @@ impl Terminal {
         }
     }
 
-    fn ask_new_todo(&mut self) -> Todo {
-        println!("Write your new todo:");
-        Todo::new(self.input())
+    fn ask_new_todo(&mut self) -> Result<Option<Todo>, TerminalError> {
+        if !self.user_intention()? {
+            return Ok(None);
+        }
+
+        self.write_stdout("Write your new todo:")?;
+        let user_input = self.input()?;
+
+        if user_input.is_empty() {
+            self.write_stdout("Please input a valid todo.")?;
+            self.ask_new_todo()
+        } else {
+            Ok(Some(Todo::new(user_input)))
+        }
     }
 
-    fn show_todo(&mut self, todo: &Todo) {
-        writeln!(self.stdout, "[ ] - {}", todo.message).unwrap()
+    fn show_todo(&mut self, todo: &Todo) -> Result<(), TerminalError> {
+        self.write_stdout(&format!("[ ] - {}", todo.message).to_string())
     }
 
-    fn user_intention(&mut self) -> bool {
-        println!("Do you want to input a new todo? (y/n)");
-        self.input() == "y"
+    fn user_intention(&mut self) -> Result<bool, TerminalError> {
+        self.write_stdout("Do you want to input a new todo? (y/n)")?;
+        let user_input = self.input();
+        Ok(matches!(user_input, Ok(input) if input == "y"))
     }
 
-    fn input(&mut self) -> String {
+    fn input(&mut self) -> Result<String, TerminalError> {
         let mut buf = String::new();
-        self.stdin.read_line(&mut buf).unwrap();
-        buf.trim().to_string()
+        self.stdin
+            .read_line(&mut buf)
+            .map_err(|error| TerminalError::Stdin(error))
```

Comment:
```suggestion
            .map_err(TerminalError::Stdin)
```

O clippy não avisou disso?